lib/src/bindings/audio_data_web.dart

```diff
@@ -0,0 +1,75 @@
+// ignore_for_file: public_member_api_docs
+
+import 'package:flutter_soloud/src/bindings/audio_data.dart';
+import 'package:flutter_soloud/src/bindings/audio_data_extensions.dart';
+import 'package:flutter_soloud/src/bindings/js_extension.dart';
+import 'package:flutter_soloud/src/bindings/soloud_controller.dart';
+import 'package:flutter_soloud/src/enums.dart';
+
+class AudioDataCtrl {
+  late final int _samplesPtr;
+  int get samplesPtr => _samplesPtr;
+
+  final void Function(AudioData) waveCallback =
+      SoLoudController().soLoudFFI.getWave;
+
+  final void Function(AudioData) texture2DCallback =
+      SoLoudController().soLoudFFI.getAudioTexture2D;
+
+  final void Function(AudioData) textureCallback =
+      SoLoudController().soLoudFFI.getAudioTexture;
+
+  final void Function(AudioData) captureWaveCallback =
+      SoLoudController().captureFFI.getCaptureWave;
+
+  final CaptureErrors Function(AudioData) captureTexture2DCallback =
+      SoLoudController().captureFFI.getCaptureAudioTexture2D;
+
+  final void Function(AudioData) captureAudioTextureCallback =
+      SoLoudController().captureFFI.getCaptureAudioTexture;
+
+  void allocSamples() {
+    _samplesPtr = wasmMalloc(512 * 256 * 4);
```

Comment:
Here I allocate the max amount of bytes needed by `*texture2D*()` and used that memory for all kind of audio samples we want to acquire. I found not convenient to allocate the memory depending of the type we want to acquire (ie the `getAudioTexture()` just needs 256*4 bytes).

Reply:
I think this should be commented in the code. If I saw `512 * 256 * 4` without any other context, I wouldn't know what this means, and would be afraid to touch it.

Also, as an aside, I suggest we find ways to remove things from the package's API surface. There's a lot of work here that you need to do just in order to maintain `AudioData`, and I don't think it's a feature that many people use. Of course, as soon as you have a feature, _some_ people will start using it, and then it's really hard to remove it because you don't want to disappoint / anger folks.

Reply:
I would like to think that this plugin is not only useful for game developers but also be used for example for a music player. In 2019 when I released my music app, my greatest enjoyment was providing users with an audio visualizer in OpenGL!
IMHO giving developers a way to capture audio data could result in something niche but also something unique. In your game, for example, you could think of making the vertices of the widget at the bottom left move based on system sounds (shots, movements, etc.). 
Not much more than what is already implemented can be added and as in the case of waveforms, we could leave the state of this feature frozen.

![filip](https://github.com/alnitak/flutter_soloud/assets/192827/7536ad6e-7acf-41c9-9eb6-7b70e048312e)

Reply:
No yeah, I get it. I'm just comparing this with the massive use case of "simply" playing audio. If you're fine with supporting this, of course I'm fine with it, too. Just please tell us long in advance if you ever feel like you start burning out of maintaining this project. It's "only" been 1 year. There are many years to come. :)